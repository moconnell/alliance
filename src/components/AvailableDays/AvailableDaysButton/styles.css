.dot {
    height: 2rem;
    width: 2rem;
    background-color: transparent;
    border-radius: 50%;
    display: flex;
    flex-direction: row;
    border: #a0a096 solid thin;
}

.clicked-dot {
    height: 2rem;
    width: 2rem;
    background-color: white;
    border-radius: 50%;
    display: flex;
    flex-direction: row;
    border: white solid thin;
}

.name-wrapper {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    margin-right: .5rem;
    margin-left: .5rem;
    color: #a0a096;
}

.name-wrapper-clicked {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    margin-right: .5rem;
    margin-left: .5rem;
    color: white;
}

.circle-wrapper {
    display: flex;
    color: white;
}