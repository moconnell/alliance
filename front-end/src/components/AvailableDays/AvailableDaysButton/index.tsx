import "./styles.css";
import { useEffect, useState } from "react";
import DaysOfWeek from "../../../types/daysOfWeek";

interface AvailableDaysButtonProps {
  day: DaysOfWeek;
  selected?: boolean;
  onSelect?: (day: DaysOfWeek, selected: boolean) => void;
}

const AvailableDaysButton = ({
  day,
  selected = false,
  onSelect,
}: AvailableDaysButtonProps) => {
  const [active, setActive] = useState(selected);

  useEffect(() => {
    setActive(selected);
  }, [selected]);

  const handleClick = () => {
    const newActive = !active;
    setActive(newActive);
    if (onSelect) onSelect(day, newActive);
  };

  return (
    <div className="circle-wrapper" role="button" aria-pressed={active} onClick={handleClick}>
      <div className={active ? "name-wrapper-clicked" : "name-wrapper"}>
        <span className={active ? "clicked-dot" : "dot"}></span>
        {DaysOfWeek[day].substring(0, 3)}
      </div>
    </div>
  );
};

export default AvailableDaysButton;
