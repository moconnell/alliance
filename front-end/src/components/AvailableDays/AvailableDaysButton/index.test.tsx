import { fireEvent, render } from "@testing-library/react";
import DaysOfWeek from "../../../types/daysOfWeek";
import AvailableDaysButton from "./index";

describe("<AvailableDaysButton />", () => {
  it("should render correctly", async () => {
    const { asFragment } = render(
      <AvailableDaysButton day={DaysOfWeek.Monday} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("should send value on click", async () => {
    const day = DaysOfWeek.Monday;
    const selected = true;
    const mockHandler = jest.fn();

    const { getByRole } = render(
      <AvailableDaysButton
        day={day}
        onSelect={mockHandler}
        selected={selected}
      />
    );

    fireEvent.click(getByRole("button"));

    expect(mockHandler).toBeCalledWith(day, !selected);
  });

  it("should ignore click without handler", async () => {
    const { getByRole } = render(
      <AvailableDaysButton day={DaysOfWeek.Monday} />
    );
    fireEvent.click(getByRole("button"));
  });
});
