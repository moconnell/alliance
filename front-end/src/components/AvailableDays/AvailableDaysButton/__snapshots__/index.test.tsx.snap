// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AvailableDaysButton /> should render correctly 1`] = `
<DocumentFragment>
  <div
    aria-pressed="false"
    class="circle-wrapper"
    role="button"
  >
    <div
      class="name-wrapper"
    >
      <span
        class="dot"
      />
      Mon
    </div>
  </div>
</DocumentFragment>
`;
